google/cloud/spanner/connection.h

```diff
@@ -54,9 +55,27 @@ class Connection {
     KeySet keys;
     std::vector<std::string> columns;
     ReadOptions read_options;
+
+    // TODO(#307): Refactor once correct location for session implemented.
+    google::cloud::optional<std::string> session_name;
```

Comment:
This will probably need the partition_token as well.

Reply:
Done.